packages/fulfillment/src/services/fulfillment-module-service.ts

```diff
@@ -671,7 +748,117 @@ export default class FulfillmentModuleService<
   ): Promise<
     FulfillmentTypes.ShippingOptionDTO[] | FulfillmentTypes.ShippingOptionDTO
   > {
-    return []
+    const updatedShippingOptions = await this.updateShippingOptions_(
+      data,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<
+      FulfillmentTypes.ShippingOptionDTO | FulfillmentTypes.ShippingOptionDTO[]
+    >(updatedShippingOptions, {
+      populate: true,
+    })
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  async updateShippingOptions_(
+    data:
+      | FulfillmentTypes.UpdateShippingOptionDTO[]
+      | FulfillmentTypes.UpdateShippingOptionDTO,
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<TShippingOptionEntity | TShippingOptionEntity[]> {
+    const dataArray = Array.isArray(data) ? data : [data]
+
+    if (!dataArray.length) {
+      return []
+    }
+
+    const shippingOptionIds = dataArray.map((s) => s.id)
+    if (!shippingOptionIds.length) {
+      return []
+    }
+
+    const shippingOptions = await this.shippingOptionService_.list(
+      {
+        id: shippingOptionIds,
+      },
+      {
+        relations: ["rules"],
+      },
+      sharedContext
+    )
+
+    FulfillmentModuleService.validateMissingShippingOptions_(
+      shippingOptions,
+      dataArray
+    )
+
+    const ruleIdsToDelete: string[] = []
+    dataArray.forEach((shippingOption) => {
+      if (!shippingOption.rules) {
+        return
+      }
+
+      const existingShippingOption = shippingOptions.find(
+        (s) => s.id === shippingOption.id
+      )!
+      const existingRules = existingShippingOption.rules
+
+      FulfillmentModuleService.validateMissingShippingOptionRules(
+        existingShippingOption,
+        shippingOption
+      )
+
+      const existingRulesMap = new Map(
+        existingRules.map((rule) => [rule.id, rule])
+      )
+      const updatedRules = shippingOption.rules
+      const toDeleteRuleIds = getSetDifference(
```

Comment:
question: Could we just use `arrayDifference` instead of converting them to sets?

Reply:
yes def 👍